services/httpd/handler.go

```diff
@@ -517,6 +518,10 @@ func (h *Handler) serveWrite(w http.ResponseWriter, r *http.Request, user *meta.
 	}
 
 	h.statMap.Add(statPointsWrittenOK, int64(len(points)))
+
+	if w.Header().Get("Content-Encoding") == "gzip" {
```

Comment:
I was about to say this won't work because the response may already of started streaming, but in this special case (`204`) it won't have, so yeah, good idea!